apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from "vscode";
+
+const typeGenPatternKey = "*.ts";
+const typeGenPattern = "${capture}.typegen.ts";
+const hasTypeGenPattern = (fileNestingPatterns: object) =>
+  Object.values(fileNestingPatterns).some((pattern: string) => {
+    if (pattern.includes(",")) {
+      const patterns = pattern.split(",");
+      return patterns.some(
+        (innerPattern) => innerPattern.trim() === typeGenPattern
+      );
+    } else {
+      return pattern === typeGenPattern;
+    }
+  });
+
+export const checkTypegenNestingConfiguration = () => {
+  const fileNestingConfig = vscode.workspace.getConfiguration(
+    "explorer.fileNesting"
+  );
```

Comment:
I wonder what this would return in pre-1.67. There is a chance that this might be undefined? or some empty obj? It would be nice if we'd make sure that we don't end up crashing in older VScode versions